akka-docs/src/main/paradox/scala/stream/stream-composition.md

```diff
@@ -197,43 +234,63 @@ needs to return a different object that provides the necessary interaction capab
 
 Next, we create a composite `Flow` from two smaller components. Here, the second enclosed `Flow` has a
-materialized type of `Future[OutgoingConnection]`, and we propagate this to the parent by using `Keep.right`
+materialized type of @scala[`Future[OutgoingConnection]`] @java[`CompletionStage<OutgoingConnection>`], and we propagate this to the parent by using `Keep.right`
 as the combiner function (indicated by the color *yellow* on the diagram):
 
-@@snip [CompositionDocSpec.scala]($code$/scala/docs/stream/CompositionDocSpec.scala) { #mat-combine-2 }
+Scala
+:   @@snip [CompositionDocSpec.scala]($code$/scala/docs/stream/CompositionDocSpec.scala) { #mat-combine-2 }
+
+Java
+:   @@snip [CompositionDocTest.java]($code$/java/jdocs/stream/CompositionDocTest.java) { #mat-combine-2 }
 
 As a third step, we create a composite `Sink`, using our `nestedFlow` as a building block. In this snippet, both
 the enclosed `Flow` and the folding `Sink` has a materialized value that is interesting for us, so
 we use `Keep.both` to get a `Pair` of them as the materialized type of `nestedSink` (indicated by the color
 *blue* on the diagram)
 
-@@snip [CompositionDocSpec.scala]($code$/scala/docs/stream/CompositionDocSpec.scala) { #mat-combine-3 }
+Scala
+:   @@snip [CompositionDocSpec.scala]($code$/scala/docs/stream/CompositionDocSpec.scala) { #mat-combine-3 }
+
+Java
+:   @@snip [CompositionDocTest.java]($code$/java/jdocs/stream/CompositionDocTest.java) { #mat-combine-3 }
 
 As the last example, we wire together `nestedSource` and `nestedSink` and we use a custom combiner function to
 create a yet another materialized type of the resulting `RunnableGraph`. This combiner function just ignores
-the `Future[Sink]` part, and wraps the other two values in a custom case class `MyClass`
+the @scala[`Future[Sink]`] @java[`CompletionStage<Sink>`] part, and wraps the other two values in a custom case class `MyClass`
 (indicated by color *purple* on the diagram):
 
-@@snip [CompositionDocSpec.scala]($code$/scala/docs/stream/CompositionDocSpec.scala) { #mat-combine-4 }
+Scala
+:   @@snip [CompositionDocSpec.scala]($code$/scala/docs/stream/CompositionDocSpec.scala) { #mat-combine-4 }
+
+Java
+:   @@snip [CompositionDocTest.java]($code$/java/jdocs/stream/CompositionDocTest.java) { #mat-combine-4a }
+    
+    @@snip [CompositionDocTest.java]($code$/java/jdocs/stream/CompositionDocTest.java) { #mat-combine-4b }
+
 
 @@@ note
 
 The nested structure in the above example is not necessary for combining the materialized values, it just
-demonstrates how the two features work together. See @ref:[Combining materialized values](stream-flows-and-basics.md#flow-combine-mat) for further examples
+demonstrates how the two features work together. See @scala[@ref:[Combining materialized values](stream-flows-and-basics.md#flow-combine-mat)] @java[@ref:[Operator Fusion](stream-flows-and-basics.md#operator-fusion)]for further examples
```

Comment:
@raboof Going forward(once the merge is completed) it won't be but as status quo there is difference...
shall i do the following:
  1. have only one reference for java/scala
  2. merge the stream-flows-and-basics.md in following commit?
Lemme know what you think